### DISCUSSION TITLE
az cli extension (bicep) does not persist to image after Packer build on Linux image

### DISCUSSION THREAD

**Discussion by rmurphyaxa** [actions/virtual-environments#6516] [General questions] [answered]
### Description
Morning,
I patched the linux image to install the az bicep cli extension after the az cli was installed. The inline script installs the az bicep cli and then verifies it's existence.
{
  "execute_command": "sudo sh -c '{{ .Vars }} {{ .Path }}'",
  "inline": [
    "echo 3_azBicepInstall",
    "az bicep install",
    "az bicep version"
  ],
  "type": "shell"
},
---
This results in the following output;

azure-arm: Provisioning with shell script: /tmp/packer-shell2117389470
azure-arm: 3_azBicepInstall
azure-arm: Installing Bicep CLI v0.12.1...
azure-arm: Successfully installed Bicep CLI to "/root/.azure/bin/bicep".
azure-arm: Bicep CLI version 0.12.1 (e43d137c2d)
azure-arm:

When the image is created and a new VM is spun up, the image no longer contains the az bicep cli.

### Platforms affected

- [X] Azure DevOps
- [ ] GitHub Actions - Standard Runners
- [ ] GitHub Actions - Larger Runners

### Runner images affected

- [ ] Ubuntu 18.04
- [X] Ubuntu 20.04
- [ ] Ubuntu 22.04
- [ ] macOS 10.15
- [ ] macOS 11
- [ ] macOS 12
- [ ] Windows Server 2019
- [ ] Windows Server 2022

### Image version and build link

ubuntu20

### Is it regression?

no

### Expected behavior

az bicep cli to be available to use

### Actual behavior

az bicep cli to be available to use

### Repro steps

Add the following provisioner step after the az cli is installed (https://github.com/actions/runner-images/blob/main/images/linux/ubuntu2004.json#L192)

{
  "execute_command": "sudo sh -c '{{ .Vars }} {{ .Path }}'",
  "inline": [
    "echo 3_azBicepInstall",
    "az bicep install",
    "az bicep version"
  ],
  "type": "shell"
},

---
The above inline scipt was added at different stages throughout the Packer file but all result in the same problem.

**Comment by al-cheb** [comment, marked as answer]
https://github.com/actions/runner-images/blob/042094639142bab5548109945dc260e5ec002ad0/images/linux/ubuntu2004.json#L381

`deprovision+user` [Step 1: Deprovision the VM](https://learn.microsoft.com/en-us/azure/virtual-machines/linux/capture-image#step-1-deprovision-the-vm) deletes machine-specific files and data. The `+user` parameter also removes the last provisioned user account. To keep user account credentials in the VM, use only -deprovision.

You have 3 options:
1. remove +user
2. install bicep globally - https://learn.microsoft.com/en-us/azure/azure-resource-manager/bicep/install
```
# Fetch the latest Bicep CLI binary
curl -Lo bicep https://github.com/Azure/bicep/releases/latest/download/bicep-linux-x64
# Mark it as executable
chmod +x ./bicep
# Add bicep to your PATH (requires admin)
sudo mv ./bicep /usr/local/bin/bicep
# Verify you can now access the 'bicep' command
bicep --help
# Done!
```
3. copy $HOME/.azure folder to /etc/skel

**Reply by rmurphyaxa** [reply]
Hi @al-cheb ,
Thanks for your suggestion but none of the above options work to keep the 'az bicep' cli installed on the linux machine after the Packer provisioning.

Option 1.
No effect and 'az bicep' not available.

Option2.
This will install the 'bicep' cli which does not contain the same functionality as the 'az bicep' cli.  The bicep cli is also installed at this point, https://github.com/actions/runner-images/blob/main/images/linux/ubuntu2004.json#L195.

Option3.
Moved the .azure with the installed az bicep cli to the /etc/skel, however, when the next vmss agent starts the az bicep cli is not available.
